js/src/hooks/useAutoCreateAdsMCAccounts.js

```diff
@@ -0,0 +1,167 @@
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+
+	const hasExistingMCAccount =
+		hasGoogleMCConnection || existingMCAccounts?.length > 0;
+	const hasExistingAdsAccount =
+		hasGoogleAdsConnection || existingAdsAccounts?.length > 0;
+
+	const googleAdsAccountChecksResolved =
+		hasFinishedResolutionForExistingAdsAccount &&
+		hasFinishedResolutionForGoogleAdsAccount;
+
+	const googleMCAccountChecksResolved =
+		hasFinishedResolutionForGoogleMCAccount &&
+		hasFinishedResolutionForExistingMCAccounts;
+
+	const accountCreationChecksResolved =
+		googleAdsAccountChecksResolved && googleMCAccountChecksResolved;
+
+	if ( accountCreationChecksResolved ) {
+		if ( ! hasExistingAdsAccount || ! hasExistingMCAccount ) {
+			const createBothAccounts =
+				! hasExistingAdsAccount && ! hasExistingMCAccount;
+
+			if ( createBothAccounts ) {
+				shouldCreateAccounts.current = CREATING_BOTH_ACCOUNTS;
+			} else if ( ! hasExistingAdsAccount ) {
+				shouldCreateAccounts.current = CREATING_ADS_ACCOUNT;
+			} else {
+				shouldCreateAccounts.current = CREATING_MC_ACCOUNT;
+			}
+		}
+	}
+
+	const handlePostAccountCreation = useCallback( () => {
+		if ( ! isCreatingWhichAccount ) {
+			return;
+		}
+
+		const mcAccountCreated = [ 200, 403, 406, 503 ].includes(
+			response?.status
+		);
+
+		const resetState =
+			( isCreatingWhichAccount === CREATING_ADS_ACCOUNT && ! loading ) ||
+			( isCreatingWhichAccount === CREATING_MC_ACCOUNT &&
+				mcAccountCreated ) ||
+			( isCreatingWhichAccount === CREATING_BOTH_ACCOUNTS &&
+				mcAccountCreated &&
+				! loading );
+
+		if ( resetState ) {
+			shouldCreateAccounts.current = null;
+			setAccountsState( ( prevState ) => ( {
+				...prevState,
+				isCreatingWhichAccount: null,
+				accountsCreated: true,
+			} ) );
+		}
+	}, [ response, loading, isCreatingWhichAccount ] );
+
+	const handleAccountCreation = useCallback( async () => {
+		if (
+			! accountCreationChecksResolved ||
+			isCreatingWhichAccount ||
+			accountsCreated
+		) {
+			return;
+		}
+
+		if ( shouldCreateAccounts.current ) {
+			setAccountsState( ( prevState ) => ( {
+				...prevState,
+				isCreatingWhichAccount: shouldCreateAccounts.current,
+			} ) );
+
+			if ( shouldCreateAccounts.current === CREATING_BOTH_ACCOUNTS ) {
+				await handleCreateAccount();
+				await upsertAdsAccount();
+			} else if (
+				shouldCreateAccounts.current === CREATING_ADS_ACCOUNT
+			) {
+				await upsertAdsAccount();
+			} else {
+				await handleCreateAccount();
+			}
+		}
+	}, [
+		accountCreationChecksResolved,
+		isCreatingWhichAccount,
+		accountsCreated,
+		handleCreateAccount,
+		upsertAdsAccount,
+	] );
+
+	useEffect( () => {
+		handlePostAccountCreation();
+	}, [ response, loading, handlePostAccountCreation ] );
+
+	useEffect( () => {
+		handleAccountCreation();
+	}, [ handleAccountCreation ] );
+
+	return {
+		hasFinishedResolutionForExistingAdsMCAccounts:
+			accountCreationChecksResolved,
```

Comment:
This state is not pretty ideal because it flashes the loading indicator to the whole card momentarily.

https://github.com/user-attachments/assets/b489ff25-bda5-4d8c-89de-5a10ed771477